inst/@sym/sym.m

```diff
@@ -189,179 +188,224 @@
+    check = false;
+    ima = ~isreal (x);
+    if (ima)
+      xx = {real(x); imag(x)};
+    else
+      xx = {x};
+    end
+    ss = cell(2,1);
+
+    for n = 1:numel(xx)
+      tmpx = xx{n};
+      [ss{n}, flag] = const_to_python_str (tmpx);
+      if (~flag)
+        % Allow 1/3 and other "small" fractions.
+        % Personally, I like a warning here so I can catch bugs.
+        % Matlab SMT does this (w/o warning).
+        % FIXME: could have sympy do this?  Or just make symbolic floats?
+        warning('OctSymPy:sym:rationalapprox', ...
+                'Using rat() heuristics for double-precision input (is this what you wanted?)');
+        [N1, D1] = rat (tmpx);
+        [N2, D2] = rat (tmpx / pi);
+        if (10*abs (D2) < abs (D1))
+          % use frac*pi if demoninator significantly shorter
+          ss{n} = sprintf ('Rational(%s, %s)*pi', num2str (N2), num2str (D2));
+        else
+          ss{n} = sprintf ('Rational(%s, %s)', num2str (N1), num2str (D1));
+        end
       else
-        s = sprintf('Rational(%s, %s)', num2str(N1), num2str(D1));
+        ss{n} = sprintf ('S(%s)', ss{n});
       end
     end
-    s = sym(s);
-    return
 
-  elseif (islogical (x)  &&  isscalar(x)  &&  nargin==1)
-    if (x)
-      cmd = 'z = sp.S.true';
+    if (ima)
+      x = sprintf('%s + I*(%s)', ss{1}, ss{2});
     else
-      cmd = 'z = sp.S.false';
+      x = ss{1};
     end
 
-  elseif (nargin == 2 && ischar(varargin{1}) && strcmp(varargin{1},'clear'))
-    % special case for 'clear', because of side-effects
-    if (isa(x, 'sym'))
-      x = x.flat;    % we just want the string
-    end
-    s = sym(x);
-    % ---------------------------------------------
-    % Muck around in the caller's namespace, replacing syms
-    % that match 'xstr' (a string) with the 'newx' sym.
-    xstr = x;
-    newx = s;
-    context = 'caller';
-    % ---------------------------------------------
-    S = evalin(context, 'whos');
-    evalin(context, '[];');  % clear 'ans'
-    for i = 1:numel(S)
-      obj = evalin(context, S(i).name);
-      [newobj, flag] = symreplace(obj, xstr, newx);
-      if flag, assignin(context, S(i).name, newobj); end
+  elseif (islogical (x)) % Handle logical values
+    check = false;
+    if (x)
+      x = 'S.true';
+    else
+      x = 'S.false';
     end
-    % ---------------------------------------------
-    return
 
-  elseif (isa (x, 'sym')  &&  (nargin >= 2))
-    % support sym(x, assumption) for existing sym x
-    s = sym(x.flat, varargin{:});
-    return
+  elseif (isinteger (x)) % Handle integer vealues
+    check = false;
+    x = num2str (x, '%ld');
+  end
 
+  if (isa (x, 'char'))
+    % We now have a char; need to decide whether to use S() or Symbol() on it.
+
+    if (check)
+      % TODO: Warning if you try make a sym with the same name of a system function.
+      %symsnotfunc (x);
+
+      % TODO: tests pass without this?  Is there a example where this is needed?
```

Comment:
Hi, i don't know how do this test, basically, if you do this ```sym(-1)``` should not send warning, but i think we can do that in other pr, only lets hold this for the future (warning when we do ops from sym)

Thx. Cya.